osquery/tables/system/darwin/xprotect.cpp

```diff
@@ -147,23 +147,32 @@ QueryData genXProtectReports(QueryContext& context) {
   return results;
 }
 
+// returns boolean indicating whether a valid plist was found
+bool findAndParsePlist(fs::path plistPath, pt::ptree& tree) {
```

Comment:
@theopolis I'm interested if you think this additional method is a net win or not, it does consolidate some of the duplicated code in `genXProtectEntries` and `getXProtectMeta`, but it isn't a huge reduction in code.

Reply:
I like it!